### DISCUSSION TITLE
✅ 🔥 Elementor v3.25 Beta Release 🔥

### DISCUSSION THREAD

**Discussion by avivu** [elementor/elementor#28960] [Announcement]
===========
# [Beta 4 UPDATED - PLEASE READ BELOW!](https://github.com/orgs/elementor/discussions/28960#discussioncomment-11027525)
===========

<img width="1200" alt="cover" src="https://github.com/user-attachments/assets/417e2a44-c106-48bb-8702-d5bdd48a302f">

&nbsp;

**Hey Beta testers,** 👋

We’re excited to announce that Elementor 3.25 Beta is now live! This release marks an important milestone in optimizing Elementor’s performance, continuing our focus on reducing page load times, improving CSS handling, and enhancing both the user and visitor experience. We’ve introduced key improvements that reduce the complexity of your website’s code, streamline search functionality, and offer native CSS solutions for smoother scrolling and better performance.

Make an impact by testing Elementor 3.25 beta and sharing your impressions on the latest additions, including:

- **Improved Performance: Continuous CSS Optimization**
- **From AnchorJS to Native CSS: New Anchor Offset**
- **[Pro] Search Widget Update: Pagination for Live Results**

Looking forward to hearing your thoughts and feedback on this update!

&nbsp;

# Improved Performance: Continuous CSS Optimization

![Improved Performance](https://github.com/user-attachments/assets/28b3b9d3-0b8d-4eb5-85cd-fa20329fb4e1)

In the previous version, Elementor 3.24, we introduced a major improvement to Conditional CSS Loading, reducing the amount of unused CSS loaded on your websites. This optimization focused on loading only the styles necessary for the specific widgets used on a page, reducing the CSS size by up to 668kb. With Elementor 3.25, we’ve taken this feature even further, introducing additional improvements to reduce the CSS footprint of Elementor pages.

### New CSS Enhancements

- **Zero CSS Weight for the Elementor Pro plugin:** Elementor Pro now starts with zero CSS by default, meaning that no styles are loaded unless they are needed on the page. This approach drastically reduces the CSS size of your website, making it faster and more efficient.
- **WooCommerce Widget Styles:** Previously, when using WooCommerce widgets, styles were loaded for all 32 widgets, even if only one was being used. In Elementor 3.25, these styles now load conditionally, saving up to 197kb of CSS, significantly improving the performance of WooCommerce-powered websites.
- **Elementor Widget Styles:** We’re advancing our efforts to conditionally load widget styles with Elementor 3.25. This update now loads styles individually for 6 Theme Elements widgets, 3 Carousel widgets, and 2 Form widgets. Instead of loading an entire group of styles at once, each widget’s styles are now loaded separately, only when that specific widget is used.
- **Global Styles:** Elementor has also optimized the way Global Styles are loaded. These styles now load conditionally, **saving up to 65kb** of CSS per page, contributing to faster loading times.

### Create Fast-Loading Websites

These improvements ensure that Elementor websites load only the CSS they need, reducing the overall page size and improving performance metrics.

By optimizing the way CSS is loaded, we’re making Elementor sites faster and more competitive in terms of SEO and user experience; Faster page load times contribute to better search engine rankings, as performance is a critical factor in how search engines like Google evaluate websites. Moreover, with less unused CSS and faster interactivity provide a smoother, more enjoyable experience for visitors, which can lead to higher engagement, lower bounce rates, and better overall conversion rates.

&nbsp;

# From AnchorJS to Native CSS: New Anchor Offset

![New Anchor Offset option](https://github.com/user-attachments/assets/d15740d3-e5e2-4440-9487-a03d562787c6)

So far we used JavaScript (JS) to create smooth scrolling effects, and while JS is effective, it has its drawbacks, including additional performance overhead and potential inconsistencies in behavior across devices. In Elementor 3.25, we’ve replaced the AnchorJS library with a native CSS solution, enabling smooth scrolling to be handled natively within the browser and ensuring faster, smoother, and more efficient scrolling.

### [Pro] New Anchor Offset Control

The transition to CSS allowed us to give you more control over how anchor links behave on your page, by introducing the new Anchor Offset control. One common issue with anchor links is that they tend to scroll to a position underneath sticky headers, hiding content. With the new Anchor Offset, anchor links can be set to scroll to the desired and correct position, ensuring all content is visible.

### Performance and User Gains

- **Enhanced Visitor Experience:** Anchor links now scroll precisely to the desired location, preventing issues where content was hidden behind sticky headers, making navigation clearer and more intuitive.
- **Cross-Device Consistency:** Native CSS ensures that smooth scrolling works consistently across browsers and devices without the potential quirks of JS-based solutions.
- **Performance Boost:** By eliminating JS from the scrolling process, we’ve reduced the load on your website’s resources and the rendering time, resulting in improved performance.

#### How to Test It:

**Check Anchor Link Functionality**

1. Create a new page with a top container that includes links.
2. Create corresponding sections below the main section, ensuring you connect the links in the top container to the sections, thus creating anchor links.
3. Ensure that the smooth scroll to the anchor works as expected.

**Check the New Anchor Offset Control**

1. Based on the layout you created earlier, go to the top container and set it to "Sticky."
2. Under the sticky control, navigate to the Anchor Offset control and set the desired offset.
3. Check the functionality of the anchor links with the sticky container.
4. Validate that the anchor links stop at the desired offset you set earlier.
5. Ensure everything works as expected in the editor and on the front end.

&nbsp;

# [Pro] Search Widget Update: Pagination for Live Results

![Pagination for Live Results](https://github.com/user-attachments/assets/a300267b-0384-4e12-8d80-fd656191b5be)

Searching for content on your site is a crucial part of the visitor experience, and with Elementor 3.25, we’ve made it even better by introducing pagination for Live Results.

Previously, in the Search widget, when search results exceeded the number of items displayed in the Live Results (e.g., showing 3 items when there are actually 10 relevant items), users couldn’t see additional results, but had to be redirected to the search results archive and navigate through the results there.

With the new pagination feature in Elementor 3.25, when a visitor performs a search, if the search returns more results than can be displayed within the Live Results, pagination options will appear at the bottom or top of the widget, with numbers and/or next/previous options to navigate through. This allows visitors to continue browsing through the results as they type within the widget, without needing to navigate away. You also have the option to customize the pagination’s typography, position, spacing and alignment from the Style section, making sure it fully fits your brand look and feel.

#### How to Test It:

1. Create a new page and add the Search widget to the canvas.
2. Go to the Results section and activate the live results.
3. You can add a loop template you created previously or navigate to the Theme Builder to create a new loop item template.
4. Next, explore the different pagination options in the Additional Settings section.
5. Choose one of the options, then navigate to the Style tab and the Additional Settings section to continue styling your pagination as desired.
6. Click on "Publish" and check the live page.
7. Ensure everything is working as expected in the editor and on the front end.

&nbsp;

# Additional Updates

![Additional Updates](https://github.com/user-attachments/assets/40d71117-b687-448c-90b3-5d3c9216ab8a)

- **Feature and Experiment Updates:**
  - **Grid Container** - Merged into Flexbox Container.
  - **Flexbox Container** - Renamed to ‘Container’.
  - **Optimized Control Loading** - Active for existing sites and promoted to Stable.
  - **Nested Elements** - Active for existing sites.
  - **Editor Top Bar** - Active for existing sites.
  - **Display Conditions (for elements)** - Native in the Editor.
  - **Form Submissions** - Native in the Editor and active on all sites. The feature can be disabled from Elementor > Settings > Advanced.
- **New Optimized Markup Performance Experiment** - Removal of the first wrapper div for Button, PayPal and Stripe widgets to simplify the markup and reduce the overall DOM. Currently in Alpha status and should not be activated on production sites.
- **Twenty Fifteen Theme** - Removal of a CSS compatibility which removes the theme’s background color on pages using the “Elementor Canvas” template.

----
# More Improvements

#### Free version

* New: Introducing Optimized Markup as an Alpha experiment - Reduce DOM size by eliminating unnecessary HTML wrappers in various elements and widgets
* Tweak: Reduced the use of JavaScript for smooth scrolling by implementing a modern CSS-based solution (#13773)
* Tweak: Updated smooth scroll behavior to respect the user's reduced motion preference setting
* Tweak: Removed `elementor-button-wrapper` div from Button widget as part of the Optimized Markup experiment
* Tweak: Load styles for Floating Elements separately to enhance performance
* Tweak: Load styles for Link in Bio widgets separately to enhance performance
* Tweak: Optimized `global.css` to exclude style generation for unused widgets
* Tweak: Removed the Floating Elements modules from `module.js`
* Tweak: Added “Edit with Elementor” button to WooCommerce's new product editor
* Tweak: Merged "Build with AI" feature to version
* Tweak: Merged "Grid Container" to "Flexbox Container" feature
* Tweak: Activated "Editor Top Bar" feature for existing sites
* Tweak: Activated "Optimized Control Loading" for existing sites and promoted the feature to stable
* Tweak: Activated "Nested Elements" feature for existing sites
* Tweak: Removed compatibility CSS support for the Twenty Fifteen theme
* Tweak: Renamed the "Flexbox Container" feature to "Container" feature
* Fix: Expand Icon in the Accordion widget would not display when nested inside another Accordion widget (#24086)
* Fix: Swiper incorrectly applied the `inert` attribute to a slide in the Nested Carousel (#23039)
* Fix: Footer would overlap the Archive Posts, Portfolio, and Gallery widgets in mobile view when using the Archive template
* Fix: "Edit with Elementor AI" button is visible for unsupported file types
* Fix: Structure panel indicator color is not visible in dark mode
* Fix: Indentation issue with empty elements in the Structure panel
* Fix: "Reference a website" feature in Elementor AI failed to work in Safari 17

#### Pro version

* Tweak: Added pagination functionality to the Live Results in Search widget
* Tweak: Added Anchor Offset control to allow precise positioning when scrolling to anchor links on the page
* Tweak: Load motion effects styles only when they are in use
* Tweak: Load sticky styles only when they are in use
* Tweak: Load popup styles only when they are in use
* Tweak: Load theme builder styles only when they are in use
* Tweak: Load transition styles for Call to Action and Gallery widgets only when they are in use
* Tweak: Load styles for Form widgets separately to enhance performance
* Tweak: Load styles for WooCommerce widgets separately to enhance performance
* Tweak: Load styles for Theme Elements widgets separately to enhance performance
* Tweak: Load styles for Carousel widgets separately to enhance performance
* Tweak: Load styles for Pricing widgets separately to enhance performance
* Tweak: Load styles for Floating Elements separately to enhance performance
* Tweak: Load styles for Link in Bio widgets separately to enhance performance
* Tweak: Load minified CSS in the Loop Builder's Call to Action button in the Editor
* Tweak: Removed "Payments" module styles
* Tweak: Removed `elementor-button-wrapper` div from Payapl widget as part of the Optimized Markup experiment
* Tweak: Removed `elementor-button-wrapper` div from Stripe widget as part of the Optimized Markup experiment
* Tweak: Merged "Form Submissions" feature into the version and moved it to the Advanced tab
* Tweak: Merged "Display Conditions" into the version
* Fix: Missing translation string for the empty message in the Table of Contents widget (#27580)
* Fix: Local Storage and Session Storage were being used even when popups were not in use (#7151)
* Fix: Masonry layout is not working when pagination is set to "Load More" or "Infinite Scroll" in Loop Grid widget (#23684)
* Fix: Extra gap between widgets on the frontend when using Off-canvas widget (#27706)
* Fix: Pagination and custom queries did not work when the Posts Widget was saved as a Global Widget or used via shortcode
* Fix: Lightbox videos are missing controls and displayed at the wrong size in Media Carousel widget
* Fix: Pagination options with Individual Pagination not working as expected in Posts and Loop Grid widgets
* Fix: Justified or Masonry layouts for the Pro Gallery would not display images on the frontend when inserted via nested tabs
* Fix: Error issue in the Loop Grid widget when used in theme parts with the Taxonomy Filter
* Fix: Removed the focus ring after clicking to open the Off Canvas inside the Editor
* Fix: Background gradient colors from the dynamic color tag values are not rendering on the frontend
* Fix: Excessive database requests generated by the Loop Grid widget
* Fix: Colors from Dynamic Tags were not rendered on the frontend
* Fix: Display issue in the Menu widget caused by the 'Native Intersection Observer API

----

## How to Install the Beta Version

If you still haven't enabled beta testing in Elementor, please follow these steps:

1. Go to Elementor > Tools > Versions
2. Enable the beta tester feature
3. Go to Dashboard > Updates and update to the latest beta version.
4. This will set you up to test the new Elementor 3.25 Beta.

## Notes

As we make changes to our GitHub in an effort to keep it organized, improve the way we deliver information to you, and our ability to review your feedback, please follow the instructions below when reporting bugs, or making feature requests:

- To report a bug related to beta versions - open a new [🐞 Bug Report](https://elemn.to/gh-new-bug-report)
- For feature requests - open a new [🚀 Feature Request](https://elemn.to/gh-new-feature-request)
- For other comments specifically about this beta version - comment here

Designated time for release: `October 28th, 2024 (Estimated)`

Many thanks for your support and help!

Cheers,
_Elementor Team_

**Comment by m1ga** [comment]
> New Optimized Markup Performance Experiment - Removal of the first wrapper div for Button, PayPal and Stripe widgets to simplify the markup and reduce the overall DOM. Currently in Alpha status and should not be activated on production sites.

Why doing it like this? A different, optimized widget would be a lot easier and you can deprecate the old button widget at some point. That way you won't break old sites and you could even use the "transform" button like with the "section -> container". And people can start using the new optimized widgets once they are ready or want to try them.

**Reply by fxaviers** [reply]
I totally agree. The button widget has been in need of a revamp for a while now. There's been issue #1907 open since 2017 (yes, a whopping 7 years ago), which deals with responsive icon positioning on the button.
Instead of a simple patch, they should put a lot more attention into it.

**Reply by nicholaszein** [reply]
Hello @m1ga and @fxaviers!

Thank you for the feedback, and for joining this beta! 🙏 

As it stands right now, this is an alpha experiment that currently requires manual activation to work, and only affects the Button, PayPal, and Stripe widgets. This is just a first step into improving the markup of a few widgets that don't and won't require complete revamping. This is why it's under an experiment, and this is why it's not a complete recreation of the entire widgets.

We understand the interest in having widgets with more functionality and less markup, but this enhancement is completely different than converting Sections/Columns elements into Container elements, as those are meant to contain other widgets, and is not meant to add new features as suggested by the community, but to improve the markup when and where it can be improved.

I hope this clarifies the intent behind this experiment.

**Reply by m1ga** [reply]
> We understand the interest in having widgets with more functionality and less markup

that wasn't my point, fxaviers added the "more features" part. 

I'm just saying that using a "Button v2" instead of an experiment that is changing the normal button will make everyone to change custom CSS/JS that is using the old structure to use the new "less DOM" layout. So it could potentially break pages in 1 year or whenever the experiment is ready.
Having a "Button v2" (I did this in a custom plugin with buttons, images and text fields by removing the `elementor-widget-container` div around it, works fine) will let you activate the experiment without breaking any old page. People who want to use it can add the new widget. It still can be behind an experiment, that is not the point. Just saying that using the normal button widget could potentially break existing CSS/JS where people rely on the current structure, having a "Button v2" makes the transition way easier and controllable. 

And you could at some point add a "convert to v2" button inside the old button so it will replace v1 with v2 for users that want to switch. 

Anyways, looking forward to see the change and then I can put my custom plugin to the graveyard ;-)

**Reply by fxaviers** [reply]
Sorry if it sounded like I was asking for more or new features.

I only mentioned issue #1907 because I read somewhere (sorry I can't remember where, it's been open for so long) that no solution has been created for the problem, as it would potentially break many existing layouts.

And if I understood correctly, this is what @m1ga initial comment says. Instead of trying to do something that could potentially create problems with existing layouts, wouldn't it be better to create a new button widget, that would already solve all the existing problems and, as already mentioned, also create a way to convert the V1 buttons to V2, requiring a non-automatic change for those who want the optimizations.

**Reply by nicholaszein** [reply]
Thank you for the clarification @m1ga! 🙏 

I completely understand your worry, and I emphasize with you, as I'm used to apply custom CSS to buttons that include the wrapper that was removed. 

Following your line of thought, eventually everyone would need to migrate to the new widget, and this would still potentially cause issues for those who chose to further customize their buttons with additional CSS or JS. So at the end of the day, its practically the same thing. We can't improve the markup and reach our goals of reducing the DOM size (as highly requested by a significant part of our users) without potentially breaking some sites. We have to choose our battles. From a risk standpoint, we understand that only a small fraction of our user-base apply these customizations. These users, like yourself, have the skills to adapt their code when that happens, and can understand the value of the trade-off. This is the least risk we can take in this situation.

We are also looking forward to these changes, as they will allow us to continue to improve the editor.

In any way, we appreciate the feedback and worry. We will make sure to inform everyone when these changes take effect to help alleviate the possible conflict.

Best regards

**Reply by nicholaszein** [reply]
Hey @fxaviers,

please do read my comment above. It may clarify it further why we are choosing to implement these changes in this way.

As for the Feature Request in question, I'm afraid the information is incorrect. It wouldn't cause breaking changes as it only requires additional CSS. However, in order for us to add more functionality to these and other widgets, first we need to settle on a markup that won't be changed in the near future that can be as flexible as possible to be used in future releases. By implementing these little changes via an experiment, we can not only gather feedback, but also expedite the markup improvement process. Creating a new widget for every single widget we have would require a lot more time. In order to ease-up the upgrading process, we decided this is the best approach for this case.

In any case, thanks for raising your points, we appreciate the feedback. This helps us understand what our users expect from us! 🙏 

Kind regards

**Reply by m1ga** [reply]
Thanks for the explanation @nicholaszein. I'm looking forward to the improvements either way you implement it!

>  So at the end of the day, its practically the same thing.

No it is not :wink: If you start a "Button v2" next to the existing widget you can enable it live in 3.25.0 and no page will break. People with new pages can start use it, people who update their pages can exchanges "Button v1" with "Button v2". People who don't to do anything can keep it. 
At some point in the future you hide the old "Button v1" in the editor. You could keep it hidden forever and all old pages will continue to work and every new page will use "Button v2".

With your experiment you replace the existing widget and if people enable it it could potentially break pages.

![Bildschirmfoto_20241016_150845](https://github.com/user-attachments/assets/47c43c36-b80a-4221-bbc8-35c72dab9904)

this is my plugin (red lines) that remove the `elementor-widget-container` and I can use it in combination with the existing widget (blue). The headline widget in this case.

I'll get your point and like I've said: happy for any improvement in this field. Less DOM = :heart:

**Reply by nicholaszein** [reply]
@m1ga,

that's a good point! However, this would be a code that would have to be maintained for quite some time. I understand where you're coming from though. For example, we replaced the old Search Form with the new Search widget. But in this case it was a matter of a necessity to write a completely new widget, with completely new functionality and markup. Buttons, and a few other widgets don't require this complete rewrite, and implementing an experiment this way ensures a wider adoption, quicker. 

I'm really excited too, this was a long coming change that will allow us to elevate Elementor to new grounds, and open it up for new possibilities. 🙂

**Reply by nicholaszein** [reply]
The discussion got off track.

Let's maintain the discussion about the current beta. 

Thanks

**Comment by Joliamendi** [comment]
Great update. A quick check I found two bugs in the menu widget.

1- The Content Width control displays an empty value.

![1](https://github.com/user-attachments/assets/57d82f41-ec02-45bd-a53f-d83283e93892)

2- The pesky bug of menu widget is back again. When the mouse moves from the menu to the menu content, the menu content is closed.

https://github.com/user-attachments/assets/8fbf3c62-0e99-406e-a92b-a66d2e6c7f4b

**Reply by doronwolf** [reply]
thank you @Joliamendi we are looking into this issue

**Reply by Joliamendi** [reply]
Hi @hein-obox, @nicholaszein and @doronwolf,

Do you approve of the #2 item? Is it clear whether it will be fixed in this version or not?

thanks🙏

**Reply by doronwolf** [reply]
Hi @Joliamendi - we were able to reproduce both issues you mentioned and aiming to fix both by the end of the beta period

**Reply by Joliamendi** [reply]
Great, thank you very much @doronwolf

**Reply by Joliamendi** [reply]
Hi again @doronwolf,

Looks like both are fixed in this beta version right?

**Reply by doronwolf** [reply]
Yes, both should be working correctly in the latest beta 😊
please let us know if you experience otherwise

**Comment by Webtica** [comment]
Hi there team,

Thanks for the update, some nice things here. 
I was testing this issue again: https://github.com/elementor/elementor/issues/25091 
I post it here as the issue is still in this latests version/beta. 
The issue is closed off so cannot post my findings there. 

Is there any news on this? Where can I ask? 
I was not expecting updates before 25th of september https://github.com/elementor/elementor/issues/25091#:~:text=Don%27t%20expect%20updates%20before%20September%2023rd. 

But we are almost a month later with no update on it. 
We have Elementor active with this issue on over 150+ sites and it is affecting our search console scores and noticeably impacting our rankings. 

Thanks for the update.

**Reply by DanielCyran** [reply]
I confirm the occurrence of this problem

**Reply by nicholaszein** [reply]
Hello @Webtica and @DanielCyran!

Thank you for joining the discussion! 🙏 

The issue in question is still under review and very much alive. There are no updates about it at the moment, and it is not part of this release.

We understand the urge to ask about the resolution of the issue or an update, and we appreciate your involvement and contributions, however, I'm afraid we are unable to discuss this issue here as it is not part of this release. 

The comments for this particular issue were locked due to offensive and aggressive comments that broke our [Code of Conduct](https://github.com/elementor/elementor/blob/main/.github/CODE_OF_CONDUCT.md#unacceptable-behaviors), and will not be unlocked until we have more information to share.

We will update you when we have more information, and we ask for your comprehension and patience.

Kind regards

**Comment by efintina** [comment]
**New Anchor Offset Control.** For me, the Anchor Offset control is not visible.

**Reply by doronwolf** [reply]
Hi @efintina thank you for noticing - we are working to fix it, next release it should display correctly for all users, thank you in advance for your patience🙏

**Reply by nicholaszein** [reply]
Hi @efintina!

Welcome, and thank you for joining the discussion! 🙏 

We just released a [new beta](https://github.com/orgs/elementor/discussions/28960#discussioncomment-10960069) that should solve this issue, can you confirm?

**Reply by efintina** [reply]
Now, it's visible and work as you said! Thanks!

**Comment by portrayaloflife** [comment]
Why are we suddenly forcing everyone to use the top bar header out of nowhere?

**Reply by contrastxxx** [reply]
"I like the new top bar, but the problem is that in desktop responsiveness, it's not possible to test responsiveness, meaning you can't resize the window."

**Reply by doronwolf** [reply]
Hi @contrastxxx can you elaborate a little bit more? what do you mean by 'can't resize the window' ?

**Reply by nicholaszein** [reply]
Hi @portrayaloflife,

thank you for joining the discussion! 🙏 

Like all Experimental Features, the Editor Top Bar is meant to be merged to core, and is part of the new interface of Elementor. This [was the plan all along](https://developers.elementor.com/elementor-3-12-beta/#:~:text=on%20the%20Frontend.-,Elementor%20Interface%20Revamp,-Elementor%203.12%20marks), and it was only implemented as an experiment to ensure everything was ported over correctly. As it's now a mature component we are expanding it to be active on all Elementor installations as the last part on our roadmap to make it the default interface for the Elementor Editor as part of our program to revamp our User Interface. 

I hope this clarifies your question! 😉 

Kind regards

**Reply by portrayaloflife** [reply]
@nicholaszein Totally understand! Change is scary!

**Reply by nicholaszein** [reply]
@portrayaloflife,

Yeah, it can be scary. This is why we introduced Experimental Features, so users can try them and get familiarized with them earlier.

Thank you for understanding! 🙏

**Reply by portrayaloflife** [reply]
@nicholaszein Quick flag though noticed for Editors with access to edit content only the top bar still shows them templates, theme builder, and template parts of the page they shouldn't have access to.

**Reply by contrastxxx** [reply]
> Hi @contrastxxx can you elaborate a little bit more? what do you mean by 'can't resize the window' ?

a feature that I use daily in the new top bar has a bug, I don't know how others solve it?
https://github.com/user-attachments/assets/36d0d364-c963-4589-8135-9d9f46c1295b

@doronwolf pls help

**Reply by nicholaszein** [reply]
Thanks for the heads up @portrayaloflife! We will check what we can do about that! 🙏

**Reply by nicholaszein** [reply]
Hi @contrastxxx!

The feature you are looking for is no longer available going forward. You can resize the canvas for any responsive mode, as long as it fits in your viewport. We have no plans to include resizing of the desktop view in the new editor interface at this moment.

Kind regards

**Reply by contrastxxx** [reply]
> Hi @contrastxxx!
> 
> The feature you are looking for is no longer available going forward. You can resize the canvas for any responsive mode, as long as it fits in your viewport. We have no plans to include resizing of the desktop view in the new editor interface at this moment.
> 
> Kind regards

"I don't understand why it's not there. It's a basic feature that I use daily. It's very important to display the range between desktop and tablet."

**Reply by nicholaszein** [reply]
@contrastxxx,

you can handle this in the tablet mode. It will always appear with the smallest possible size for the tablet, and you can increase it until it reaches the largest possible size for tablets. This haven't changed.

**Reply by ghost** [reply]
>  This haven't changed.

@nicholaszein why you blocked 2 topics?  this features wants more people than others.

https://github.com/orgs/elementor/discussions/22796
https://github.com/orgs/elementor/discussions/22853

I think this is bad relationship with customers. Open this again.

**Reply by nicholaszein** [reply]
Hi @necrosoul,

Thank you for the feedback! We will take it into consideration going forward! 🙏

We understand the need to discuss other topics, but we would be delighted if you could spare some time to test this new beta and share your feedback about the improvements we introduced in this version. We want to ensure these improvements are stable and compatible with your existing sites.

As always, make sure you're testing on a staging environment to avoid possible conflicts.

Thank you again, and kind regards!

**Reply by ghost** [reply]
> Hi @necrosoul,
> 
> Thank you for the feedback! We will take it into consideration going forward! 🙏
> 
> We understand the need to discuss other topics, 

@nicholaszein i want to discuss about new version, but these 2 topics is more important for me and others. 
I dont like offtopic, but you CLOSED these topics. I cant write in these topics. So.. thats why i writed here.

**Comment by wordxpression** [comment]
With regards to the anchor offset. I can enter it in pixels. But when I would use another unit (vh, % or whatever) for the height of a sticky container, it still would not work as intended. 

Would be nice if the value could be in other units.

**Reply by doronwolf** [reply]
Hi @wordxpression thank you for the feedback 🙂 we are hoping to improve that in future version

**Comment by nicholaszein** [comment]
Hi all,

We just released **Elementor v3.25 Beta 2**


The versions include the following:


#### Free version

* Fix: Restore skip button functionality in Onboarding

#### Pro version

* Fix: The new Anchor offset control is missing


–

Thank you!

**Reply by portrayaloflife** [reply]
This all looks awesome! I noticed a lot of loop grid fixes and pagination in the change log. Really great to see the continued development on loop grid and making it rock solid! Was curious if this "Fix: Pagination options with Individual Pagination not working as expected in Posts and Loop Grid widgets" is a fix to this? >>  [#24973](https://github.com/elementor/elementor/issues/24973)

If so I am hyped! Also is the "less database requests by loop grid" going to speed up page load times for pages with larger feeds by chance? 

Appreciate all y'all's hard work!

**Reply by nicholaszein** [reply]
@portrayaloflife!

Thanks for the kind and encouraging words, it means a lot to us! ❤️ 

I'm still in the process of validating all issues related to the pagination improvements we did, so for the time being I don't have an answer to your question about that specific issue.

As for performance, fewer database requests definitely improve the page speed loading for loops in general! 😃

**Reply by doronwolf** [reply]
Hi @portrayaloflife regarding [#24973](https://github.com/elementor/elementor/issues/24973) It's a different issue than the "Fix: Pagination options with Individual Pagination not working as expected in Posts and Loop Grid widgets" but **we are aware of this issue** and aiming to fix it in future versions

**Reply by portrayaloflife** [reply]
Roger that! Thanks for being on top of it y'all! Really appreciate it!

**Comment by doronwolf** [comment]
Hi all,
We just released **Elementor v3.25 Beta 3**.
The versions include the following:

#### Free version
* Fix: General code improvements in this version

#### Pro version
* Fix: Dropdown area not working in certain scenarios in Menu widget

–

Thank you!

**Comment by lensbreak** [comment]
Will all these CSS and performance improvements finally solve the https://github.com/elementor/elementor/issues/25091 problems?

**Reply by doronwolf** [reply]
our R&D team is currently investigating the #25091 problem, 
we will try to update our findings about this issue when they are ready

**Comment by avivu** [comment]
Hi all,

We just released **Elementor Pro v3.25 Beta 4**.


The version includes only the following Pro changes:


#### Pro version


* Fix: Content Width control inside the Mega Menu is displaying an empty value
* Fix: Adjusted CSS style loading for the Form and Login widgets



–

Thank you!

**Reply by lensbreak** [reply]
Thank you, we are going to test it well.

**Comment by lensbreak** [comment]
Thank you @doronwolf I urge you to ensure that every one of your developers channels their full, undivided attention to this issue – immediately. No distractions, no delays. Elementor powers a staggering 11.4% of all websites globally, and the impact of this bug cannot be overstated.

Having used Elementor for over five years, this is by far the most critical and disruptive bug we've encountered. We manage numerous websites with Elementor, and every single one is affected by the https://github.com/elementor/elementor/issues/25091 issue. The problem is especially devastating for webstores, where it directly triggers a serious LCP problem, flagged consistently across all Google Search Consoles. This bug is wreaking havoc on SEO and page rankings.

Our in-depth analysis has revealed that the random fluctuations in web vitals, as noted in https://github.com/elementor/elementor/issues/25091, directly correlate with massive revenue losses for affected online stores since these inconsistencies are crippling the visibility of these websites, depending on unpredictable Google scans.

**This bug has persisted for over a year. The time to fix it is not tomorrow. It’s now.**

**Comment by black-eye** [comment]
Hi, in 3.25 Form Submissions were merged into the core, but they can be still disabled, just the settings moved to Elementor > Settings > Advanced. In previous version I had this feature disabled and I would expect this will be kept after the update, even in the new location. But it's enabled again. Is it a bug or by design? I would definitely prefer to keep the previous settings.

**Reply by nicholaszein** [reply]
Hi @black-eye.

This is not a bug, it should be active by default. However, I see what you mean and we will discuss this internally. Thanks for raising this concern! 🙏

**Reply by tobias992** [reply]
Dear Dev-Team,
You have no right to change the settings on my site. It is none of your business! Some of these settings have been deliberately set by admins - especially in relation to GDPR. So please, keep the settings how they are.

**Reply by portrayaloflife** [reply]
I get why y'all auto turn on these features, but I do wish there was an option for some of these features to keep off, or at the very least, elementor side more of a warning, that like X feature will be merged on X date and now always turned on. Just for planning purposes. Some of these end up being surprises. Like the Top bar switch. Lot of folks gonna be surprised after that update, vs knowing that's something they should of been migrating to from the get-go. I know it's a complex kind of thing to communicate, but just expectations around features and when/how they are being merged would be sick! Respectfully.

**Reply by black-eye** [reply]
Hi @nicholaszein. I fully understand, when some feature is turned on by default. But this feature I've already deliberately turned off (and I had a reason for it). Why should I have to turn it off again? Just FYI, I have LTD on Fluent Forms which I use on every site I build. I never use Elementor forms, thus will never use Form Submissions - no need this extra feature to be turned on (and I won't change my mind, even if you will keep turning it on for me with every update).

Frankly, I don't get, what benefit do you expect to gain from turning on the features once turned off. Same way I don't understand why do you think we will be happy, when you are removing options for disabling features. IMHO for every major feature there should be an option to turn it off.  Why you just cannot let your users decide, how to use the tool? What benefit do you expect from forcing us to use it the way you want? It doesn't make any sense - we are the ones who are using the tool, we know best, what we want to achieve. More options to customize the tool = more happiness of your users. It's that simple ;-)

**Reply by nicholaszein** [reply]
@black-eye, I think I missed the mark here, and haven't explained it correctly. Apologies for that.

The Form Submissions feature is turned on by default upon installation of Elementor Pro. The user chooses to turn it off it they don't want to use it. That's the intended behavior, otherwise people would never know it exists. In this version, there will be a Notice about this feature being merged, and how to deactivate it. 

However, as you pointed yourself, if you choose to turn it off, you expect it to continue to be off. So if it's turned on now that it was moved to its final location, we need to verify the logic, and see if we can avoid that. This is why I raised this to the team to see if we can do something about it. 

We are not, in any way, trying to force anyone to use this feature. We just want people to know it exists. It was one of the most requested features after all.

I hope this clarifies my answer, and apologies for the misunderstanding. 🙏

**Reply by tobias992** [reply]
Bravo Elementor team, you have reached a new milestone. After the update, you changed the form submission settings from disabled to active. How arrogant and conceited are you going to get? You change the settings on my site - where I am the sole master of the settings. Shame on you!

**Reply by nicholaszein** [reply]
Hi @black-eye,

due to some technical difficulties, we were unable to transfer the choices made for the Form Submissions as an experiment to its final setting.

We understand the inconvenience this may cause, and we apologize for the trouble.

Luckily this option can be deactivated by going to your WordPress **`Dashboard`** > **`Elementor`** > **`Settings`** > **`Advanced`** page.

You will also receive a Dashboard Notice explaining how you can deactivate Form Submissions if you want to.

![image](https://github.com/user-attachments/assets/8515c1b6-49b1-4b87-b5da-c46c97997bf4)

Kind regards

**Comment by black-eye** [comment]
I have a very little time to test things, but it seems that the file "uploads/elementor/css/custom-pro-frontend.min.css", which, according to https://developers.elementor.com/elementor-3-25-developers-update/, should be deleted, is still referenced in the html (even though is seems it really doesn't exist). 

Also, in the post mentioned above, the file is named "frontend.min.css", mine is named "custom-pro-frontend.min.css". TBH, I'm not sure they are even meant to be the same files. But the fact is, that I'm getting the error that this file wasn't found.

The site, I'm testing it on has quite a long history and a lot of experiments are still turned off there, so maybe this is the reason. But please check it out, just to be sure...

**Reply by hein-obox** [reply]
@black-eye Your observations are correct. We are working on a fix.

**Reply by hein-obox** [reply]
@black-eye I will be updated in the next 3.25 version.

**Comment by Mesmer99** [comment]
Hello guys. Do you plan to improve work between loop builder and off canvas widget? I've mentioned some details in https://github.com/orgs/elementor/discussions/28935. The thing is demand of dynamic preview functionality of posts,products and etc. is going on from almost 6 years ago and I'm sure based on how many times this discussed its a crucial functionality.

**Reply by doronwolf** [reply]
Hi @Mesmer99 🙂
The off-canvas was introduced as part of version 3.22 and the loop builder has been here since version 3.8 so I am not sure what you mean by '6 years ago' but I do agree users are looking for this feature and I can confirm that it's in our 'to-do' list, 
however, I can't commit to when it will be ready, thank you for creating a feature request for it - the more people will vote for it, the faster we can promote it👍

**Comment by portrayaloflife** [comment]
Random question, possible to add the AI upsell off switch to the Elementor Role Manager? As an admin I'm totally fine with AI and can utilize it, but would great to be able to shut it off for our Editors/authors across the board rather than have to go user by user. Just a thought! Appreciate all the awesome work!

**Reply by m1ga** [reply]
check the reply in the old thread: https://github.com/orgs/elementor/discussions/28508#discussioncomment-10448938 and the last one (no ETA at the moment).

**Comment by nicholaszein** [comment]
**Hey beta testers! 👋** 

### We're happy to announce that Elementor and Elementor Pro version 3.25.0 are now available for update for the general public! 🚀 🥳 🙌 

Thank you for contributing to this discussion! 🙏 

**Cheers 🥂**

> Changelogs of Elementor Core and Pro
> [Elementor Core](https://wordpress.org/plugins/elementor/#developers)
> [Elementor Pro](https://elementor.com/pro/changelog/)
